src/main/java/mate/academy/spring/controller/OrderController.java

```diff
@@ -34,14 +34,20 @@ public OrderController(ShoppingCartService shoppingCartService,
     }
 
     @PostMapping("/complete")
-    public OrderResponseDto completeOrder(@RequestParam Long userId) {
-        ShoppingCart cart = shoppingCartService.getByUser(userService.get(userId));
+    public OrderResponseDto completeOrder(Authentication authentication) {
+        ShoppingCart cart = shoppingCartService
+                .getByUser(userService.findByEmail(authentication.getName()).orElseThrow(
+                        () -> new RuntimeException("Can't find users email"
+                                + authentication.getName())));
         return orderResponseDtoMapper.mapToDto(orderService.completeOrder(cart));
     }
 
     @GetMapping
-    public List<OrderResponseDto> getOrderHistory(@RequestParam Long userId) {
-        return orderService.getOrdersHistory(userService.get(userId))
+    public List<OrderResponseDto> getOrderHistory(Authentication authentication) {
+        return orderService.getOrdersHistory(userService
+                        .findByEmail(authentication.getName()).orElseThrow(
+                                () -> new RuntimeException("Can't find users email"
+                                        + authentication.getName())))
                 .stream()
```

Comment:
It's really hard to read this code, lets first get User from db, and then get order history for him